### ISSUE TITLE
Add non regression tests

### ISSUE THREAD

**Issue by rom1504** [mlfoundations/open_clip#198]
To make it easier to merge PRs in the future, we need automated non regression tests
* In GitHub action
* on CPU 

Tests to have: 

* [x] exact inference result before/after
   * [x] non crashing inference
   * [x] non reg : wip at 199
* [ ] can still load with torch.load
* [ ] training works the same before after, for one step, same weights and results
   * [x] non crashing training  
   * [ ] non reg

**Comment by rom1504** [comment]
@Iopho if you want to help here, it would be appreciated

**Comment by lopho** [comment]
I made some time tomorrow and will start writing tests.

**Comment by lopho** [comment]
@rom1504 I open a PR draft for inference testing #199

**Comment by rom1504** [comment]
non crashing training test done at #203, non reg still to do

**Comment by rom1504** [comment]
Non reg inference tests done, thanks @lopho

Next step here is non reg test for training

**Comment by rom1504** [comment]
Still very much interested by the non reg for training if you feel like continuing work here @lopho 
Develoment is still active and if we can make sure we don't break things going forward it will be great

**Comment by lopho** [comment]
Yes, I'm still working on it. Before training I still hvae to look into making testing more stable regarding randomness between runs, as I think this will be worse with training tests.

**Comment by rom1504** [comment]
Looks like inference testing is working very well with your new strategy @lopho 

Do you want to look into training non reg as well ?

We have a few PRs with significant changes and it would be very helpful if we can be confident that base clip training is still doing exactly the same thing

**Comment by lopho** [comment]
Definetly, but I won't have time to until after new years.

**Comment by lopho** [comment]
I will start working on this again within the next week, unless there is already someone working on it that I am not aware of.

**Comment by rom1504** [comment]
@lopho would be great. We're still in need of more reliability for merging PRs!
Currently doing manual training for validation which is painful
